File: model/src/main/java/io/github/chriszhong/model/organization/function/FunctionModule.java
package io.github.chriszhong.model.organization.function;

import io.github.runtimemodels.chazm.function.Effectiveness;
import io.github.runtimemodels.chazm.function.Goodness;

import com.google.inject.AbstractModule;

/**
 * The {@linkplain FunctionModule} class provides a Guice binding module for functions.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
public class FunctionModule extends AbstractModule {

	@Override
	protected void configure() {
		bind(Effectiveness.class).to(DefaultEffectiveness.class);
		bind(Goodness.class).to(DefaultGoodness.class);
	}

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/guice/FunctionModule.kt
package runtimemodels.chazm.model.guice

import com.google.inject.AbstractModule
import runtimemodels.chazm.api.function.Effectiveness
import runtimemodels.chazm.api.function.Goodness
import runtimemodels.chazm.model.function.DefaultEffectiveness
import runtimemodels.chazm.model.function.DefaultGoodness

/**
 * The [FunctionModule] class provides a Guice binding module for functions.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
class FunctionModule : AbstractModule() {

    override fun configure() {
        bind(Effectiveness::class.java).to(DefaultEffectiveness::class.java)
        bind(Goodness::class.java).to(DefaultGoodness::class.java)
    }

}
